### PR TITLE
PATH_STANDBY/AVAILABLE can be sent for all issued DCIDs

### LINKED ISSUE #272: When to send PATH_STATUS
Is an endpoint allowed to send PATH_STATUS frames for DCID=X before it observes X being used?

I think it would be natural especially for servers to bundle a NEW_CONNECTION_ID frame of DCID=X and a PATH_STATUS frame of DCID=X. That's the easiest way of declaring that all paths can be used for sending data.

However, the draft is vague if such behavior is allowed, or if it is forbidden to send PATH_STATUS frames before receiving a packet that uses that DCID value.

Hence opening the issue.

### PR DESCRIPTION
**Pull request by mirjak**
fixes #272

### CODE REVIEW

File: draft-ietf-quic-multipath.md [hunk 1]
```diff
@@ -1154,6 +1155,10 @@ PATH_STANDBY frame is considered lost, the peer SHOULD resend the frame
 only if it contains the last status sent for that path -- as indicated
 by the sequence number.
 
+A PATH_STANDY may be bundled with a NEW_CONNECTION_ID frame or
```
Comment (mirjak) [inline draft-ietf-quic-multipath.md:1158]:
```suggestion
A PATH_STANDY MAY be bundled with a NEW_CONNECTION_ID frame or
```

File: draft-ietf-quic-multipath.md [hunk 2]
```diff
@@ -1200,6 +1206,10 @@ PATH_AVAILABLE frame is considered lost, the peer SHOULD resend the frame
 only if it contains the last status sent for that path -- as indicated
 by the sequence number.
 
+A PATH_AVAILABLE may be bundled with a NEW_CONNECTION_ID frame or
```
Comment (mirjak) [inline draft-ietf-quic-multipath.md:1209]:
```suggestion
A PATH_AVAILABLE MAY be bundled with a NEW_CONNECTION_ID frame or
```

**Review by Yanmei-Liu — APPROVED** [review]

File: draft-ietf-quic-multipath.md [hunk 3]
```diff
@@ -1130,7 +1130,8 @@ PATH_STANDBY Frames contain the following fields:
 Destination Connection ID Sequence Number:
 : The sequence number of the Destination Connection ID used by the
   receiver of this frame to send packets over the path the status update
-  corresponds to.
+  corresponds to. All Destination Connection ID that have been issued
+  MAY be used even if a path was not initated yet.
```
Comment (qdeconinck) [inline draft-ietf-quic-multipath.md:1134]:
```suggestion
  MAY be used even if it is not yet in user over a path.
```

File: draft-ietf-quic-multipath.md [hunk 4]
```diff
@@ -1154,6 +1155,10 @@ PATH_STANDBY frame is considered lost, the peer SHOULD resend the frame
 only if it contains the last status sent for that path -- as indicated
 by the sequence number.
 
+A PATH_STANDY MAY be bundled with a NEW_CONNECTION_ID frame or
```
Comment (qdeconinck) [inline draft-ietf-quic-multipath.md:1158]:
```suggestion
A PATH_STANDBY frame MAY be bundled with a NEW_CONNECTION_ID frame or
```

File: draft-ietf-quic-multipath.md [hunk 5]
```diff
@@ -1154,6 +1155,10 @@ PATH_STANDBY frame is considered lost, the peer SHOULD resend the frame
...
+PATH_RESPONSE frame in order to indicated the prefered path usage
```
Comment (qdeconinck) [inline draft-ietf-quic-multipath.md:1159]:
```suggestion
PATH_RESPONSE frame in order to indicate the preferred path usage
```

File: draft-ietf-quic-multipath.md [hunk 6]
```diff
@@ -1176,7 +1181,8 @@ PATH_AVAILABLE frames contain the following fields:
 Destination Connection ID Sequence Number:
 : The sequence number of the Destination Connection ID used by the
   receiver of this frame to send packets over the path the status update
-  corresponds to.
+  corresponds to. All Destination Connection ID that have been issued
+  MAY be used even if a path was not initated yet.
```
Comment (qdeconinck) [inline draft-ietf-quic-multipath.md:1185]:
```suggestion
  MAY be used even if it is not yet in user over a path.
```

File: draft-ietf-quic-multipath.md [hunk 7]
```diff
@@ -1200,6 +1206,10 @@ PATH_AVAILABLE frame is considered lost, the peer SHOULD resend the frame
 only if it contains the last status sent for that path -- as indicated
 by the sequence number.
 
+A PATH_AVAILABLE MAY be bundled with a NEW_CONNECTION_ID frame or
```
Comment (qdeconinck) [inline draft-ietf-quic-multipath.md:1209]:
```suggestion
A PATH_AVAILABLE frame MAY be bundled with a NEW_CONNECTION_ID frame or
```

File: draft-ietf-quic-multipath.md [hunk 8]
```diff
@@ -1200,6 +1206,10 @@ PATH_AVAILABLE frame is considered lost, the peer SHOULD resend the frame
...
+PATH_RESPONSE frame in order to indicated the prefered path usage
```
Comment (qdeconinck) [inline draft-ietf-quic-multipath.md:1210]:
```suggestion
PATH_RESPONSE frame in order to indicate the preferred path usage
```

**Review by qdeconinck — CHANGES_REQUESTED** [review]
Mostly typos.

File: draft-ietf-quic-multipath.md (hunk 3 above)
Comment (kazuho) [inline draft-ietf-quic-multipath.md:1134]:
```suggestion
  MAY be specified even if it is not yet in use over a path.
```
"specified" in addition to @qdeconinck's suggestions.

File: draft-ietf-quic-multipath.md (hunk 6 above)
Comment (kazuho) [inline draft-ietf-quic-multipath.md:1185]:
```suggestion
  MAY be specified even if it is not yet in use over a path.
```

**Review by kazuho — COMMENTED** [review]
Thank you for the PR.

Looks good modulo editorial preferences.

**Review by huitema — APPROVED** [review]
This is fine. On one hand, it forces the implementers to remember available/standby status per CID, on the other hand, it suppresses a race condition. Since we already have to maintain sequence number state per CID, this is an acceptable tradeoff.

File: draft-ietf-quic-multipath.md [hunk 9]
```diff
@@ -1154,6 +1155,10 @@ PATH_STANDBY frame is considered lost, the peer SHOULD resend the frame
 only if it contains the last status sent for that path -- as indicated
 by the sequence number.
 
+A PATH_STANDBY frame MAY be bundled with a NEW_CONNECTION_ID frame or
+PATH_RESPONSE frame in order to indicate the preferred path usage
```
Comment (mirjak) [inline draft-ietf-quic-multipath.md:1159]: Do we need to also mention PATH_CHALLENGE?

**Review by yfmascgy — COMMENTED** [review]
Some minor changes

**Review by yfmascgy — COMMENTED** [review]
Looks good to me. Some minor changes.

**Pull request merged**